File: frontend/lib/api/subscription.ts
import { apiFetch } from "./client";

export interface SubscriptionResponse {
  id: string;
  org_id: string;
  plan: string;
  status: string;
  current_period_end: string | null;
  checkout_url?: string;
  billing_interval?: string;
}


export async function getSubscriptionStatus(): Promise<SubscriptionResponse> {
  return apiFetch<SubscriptionResponse>("/api/v1/subscriptions/status");
}

export async function subscribeToPlan(plan: string, interval: "month" | "year" = "month"): Promise<SubscriptionResponse> {
  return apiFetch<SubscriptionResponse>("/api/v1/subscriptions/subscribe", {
    method: "POST",
    body: { plan, interval },
  });
}

export async function cancelSubscription(): Promise<SubscriptionResponse> {
  return apiFetch<SubscriptionResponse>("/api/v1/subscriptions/cancel", {
    method: "POST",
  });
}

export async function confirmPayment(transactionId: string, plan?: string): Promise<{ status: string; plan: string }> {
  const url = plan 
    ? `/api/v1/subscriptions/confirm-payment?transaction_id=${transactionId}&plan=${plan}`
    : `/api/v1/subscriptions/confirm-payment?transaction_id=${transactionId}`;
  return apiFetch<{ status: string; plan: string }>(url, {
    method: "POST",
  });
}

File: backend/app/schemas/subscription.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class SubscriptionResponse(BaseModel):
    id: uuid.UUID
    org_id: uuid.UUID
    plan: str
    status: str
    current_period_end: datetime | None = None
    checkout_url: str | None = None
    billing_interval: str = "month"

    model_config = {"from_attributes": True}



class SubscribeRequest(BaseModel):
    plan: str = Field(pattern="^(free|growth|pro|enterprise)$")
    interval: str = Field(default="month", pattern="^(month|year)$")

